corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js

```diff
@@ -243,7 +243,15 @@ WebFormSession.prototype.handleFailure = function (resp, action, textStatus, fai
     } else if (textStatus === 'timeout') {
         errorMessage = Formplayer.Errors.TIMEOUT_ERROR;
     } else if (resp.hasOwnProperty('responseJSON')) {
-        errorMessage = Formplayer.Utils.touchformsError(resp.responseJSON.message);
+        if (resp.responseJSON === undefined) {
```

Comment:
Is there a more direct way to check for a connectivity issue?  Like some particular response code or checking `window.navigator.onLine`?

Reply:
updated